### PR TITLE
Add WARNING for schema/schematron version mismatch

### LINKED ISSUE #628: As a user, I want to throw a WARNING when a product's schematron version does not match the schema version
### Checked for duplicates

Yes - I've already checked

### 🧑‍🔬 User Persona(s)

Data User, Archivist

### 💪 Motivation

...so that I can ensure the versions of the schematron and schema are the same.

### 📖 Additional Details

Unfortunately, the only way we can really check this is via the filenames in the `xml-model` element and `schemaLocation`, but it is better than nothing.

### Acceptance Criteria

**Given** a product label with a v1H00 schematron and a v1J00 schema
**When I perform** validation of the product
**Then I expect** a WARNING to be thrown for a version mismatch

**Given** a product label with a v1H00 schematron and a v1H00 schema
**When I perform** validation of the product
**Then I expect** the product to validate successfully

(probably only need a test case for the first acceptance criteria)

### ⚙️ Engineering Details

Test data: 
[mp2_flat_20061109.xml.txt](https://github.com/NASA-PDS/validate/files/11253671/mp2_flat_20061109.xml.txt)

### LINKED ISSUE #629: Add Check for Schematron/Schema Version Mismatch
## 💡 Description

- https://github.com/NASA-PDS/validate/issues/628

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary
Added a function to check the schema versions against the schematron versions. It only works if the same naming convention for the files themselves (not the paths to them) are the same. Warnings are given if they do not match

## ⚙️ Test Data and/or Report
See unit tests

## ♻️ Related Issues
Closes #629
Closes #628

### CODE REVIEW

File: src/main/java/gov/nasa/pds/tools/label/LabelValidator.java [hunk 1]
```diff
@@ -383,6 +383,73 @@ private Boolean determineSchematronValidationFlag(URL url) {
...
+            // Ignore error since XMLExtractor() was able to read it this should be alright
+          }
+        }
+        for (String model : specifiedSchematron) {
+          for (String part : model.split("\\s")) {
+            if (part.toLowerCase().startsWith("href")) {
+              String modelHREF = part.substring(part.indexOf('"')+1, part.lastIndexOf('"'));
+              try {
+                URL schematronURL = new URL(modelHREF);
+                String schematronName = Paths.get(schematronURL.getPath()).getFileName().toString().toLowerCase();
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply src/main/java/gov/nasa/pds/tools/label/LabelValidator.java]: @sonatype-lift ignore

File: src/main/java/gov/nasa/pds/tools/label/LabelValidator.java [hunk 2]
```diff
@@ -383,6 +383,73 @@ private Boolean determineSchematronValidationFlag(URL url) {
...
+        handler.addProblem(new ValidationProblem(
+            new ProblemDefinition(ExceptionType.WARNING, ProblemType.SCHEMATRON_WARNING,  "Cannot check versioning because no Schematron given."),
+            url));
+      } else {
+        Set<String> schemas = new HashSet<String>();
+        Set<String> schematrons = new HashSet<String>();
+        for (String schema : specifiedSchema) {
+          try {
+            URL schemaURL = new URL(schema);
+            String schemaName = Paths.get(schemaURL.getPath()).getFileName().toString().toLowerCase();
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply src/main/java/gov/nasa/pds/tools/label/LabelValidator.java]: @sonatype-lift ignore

File: src/main/java/gov/nasa/pds/tools/label/LabelValidator.java [hunk 3]
```diff
@@ -383,6 +383,73 @@ private Boolean determineSchematronValidationFlag(URL url) {
...
+        if (0 < uniqueSchema.size() + uniqueSchematron.size()) {
+          handler.addProblem(new ValidationProblem(
+              new ProblemDefinition(ExceptionType.WARNING, ProblemType.SCHEMA_WARNING,
+                  "The schema version(s) " + uniqueSchema.toString() +
+                  " does/do not match the schematron version(s) " + uniqueSchematron.toString() + "."),
+              url));
+        }
+      }
+    }
+    catch (XPathException | XPathExpressionException e) {
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/tools/label/LabelValidator.java]: Neat! I didn't know you could use `|` to catch multiple kinds of exceptions

**Review by nutjob4life — APPROVED** [review]
👍

**Comment by nutjob4life** [comment]
@al-niessner approved and ready to go, but I haven't merged this since there might be a jparser dependency. I'll let you go ahead and hit the big green button!

**Review by jordanpadams — APPROVED** [review]

**Pull request merged**